### PR TITLE
F# friendly

### PR DESCRIPTION
**Pull request by isaacabraham**
This amends the build chain to a more typical F# project:

Removes
* ~~NuGet~~
* ~~CAKE script~~
* ~~Powershell bootstrapper~~

Adds: 
* Paket
* FAKE
* Simple build bootstrapper

Hopefully this will also encourage more users from the F# community to participate in the project (as well as make life easier for the maintainers!).

### CODE REVIEW

**Comment by bricelam** [comment]
This repo just got about 20% cooler.

File: build.fsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+open Fake
+let target = getBuildParam "target"
+let configuration = getBuildParamOrDefault "configuration" "Release"
+let signOutput = hasBuildParam "signOutput"
+
+Target "Build" <| fun _ ->
+    DotNetCli.Build <| fun p ->
+        { p with
+            Configuration = configuration
+            Project = "EFCore.FSharp.sln" }
```
Comment (bricelam) [inline build.fsx]: Flow `signOutput`? (This lets me sign the builds before releasing to NuGet.org)

Comment (isaacabraham) [reply build.fsx]: Let me see how to do that :-)

Comment (isaacabraham) [reply build.fsx]: How should this flow into MSBuild? `/p:SignOutput=True` or something else?

Comment (isaacabraham) [reply build.fsx]: @bricelam ok I just pushed some extra bits to the FAKE script, this should now be working (hopefully!).

File: .gitignore [hunk 2]
```diff
@@ -286,3 +286,4 @@ tools/**
 *.btm.cs
 *.odx.cs
 *.xsd.cs
+/packages/
```
Comment (forki) [inline .gitignore]: please add /paket-files

**Comment by forki** [comment]
* /paket-files need to be gitignored
* build.cmd needs to be changed
* please add a build.sh

Optional:
* appveyor config
* travis config

**Comment by forki** [comment]
packages.config should be nuked

**Comment by isaacabraham** [comment]
Hoppla, that one must've escaped. I'll remove it.

**Comment by isaacabraham** [comment]
What does build.cmd need to be changed to? The error handler?

**Comment by forki** [comment]
never mind. just saw you fixed build.cmd as well.

**Comment by isaacabraham** [comment]
@forki All done (I think).

**Comment by simontaite** [comment]
@forki +1 for Appveyor and Travis builds
Have created #8 and #9 to track those

**Pull request merged**